### PR TITLE
tf.keras Serialization fix

### LINKED ISSUE #422: BatchNormalization layer fails to deserialize when working with Tensorflow
**Describe the bug**
When testing Tensorflow with more complex model, BatchNormalization layer fails to deserialize with error:
```
           ERROR    Collaborator failed with error: Unknown layer: BatchNormalization. Please ensure this object is passed to the `custom_objects`   envoy.py:75
                    argument. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.:
                    Traceback (most recent call last):
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/component/envoy/envoy.py", line 73, in run
                        self._run_collaborator()
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/component/envoy/envoy.py", line 138, in _run_collaborator
                        col = plan.get_collaborator(self.name, self.root_certificate, self.private_key,
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/federated/plan/plan.py", line 435, in get_collaborator
                        model_provider, task_keeper, data_loader = self.deserialize_interface_objects()
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/federated/plan/plan.py", line 541, in <genexpr>
                        return (self.restore_object(api_layer['settings'][filename]) for filename in filenames)
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/federated/plan/plan.py", line 561, in restore_object
                        obj = serializer_plugin.restore_object(filename)
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/plugins/interface_serializer/cloudpickle_serializer.py", line 27,
                    in restore_object
                        return cloudpickle.load(f)
                      File "/home/openfl/venv/lib/python3.8/site-packages/openfl/plugins/frameworks_adapters/keras_adapter.py", line 24, in unpack
                        restored_model = deserialize(model)
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/layers/serialization.py", line 123, in deserialize
                        return generic_utils.deserialize_keras_object(
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 674, in
                    deserialize_keras_object
                        deserialized_obj = cls.from_config(
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/engine/functional.py", line 667, in from_config
                        input_tensors, output_tensors, created_layers = reconstruct_from_config(
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/engine/functional.py", line 1278, in
                    reconstruct_from_config
                        process_layer(layer_data)
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/engine/functional.py", line 1260, in process_layer
                        layer = deserialize_layer(layer_data, custom_objects=custom_objects)
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/layers/serialization.py", line 123, in deserialize
                        return generic_utils.deserialize_keras_object(
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 659, in
                    deserialize_keras_object
                        (cls, cls_config) = class_and_config_for_serialized_keras_object(
                      File "/home/openfl/venv/lib/python3.8/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 556, in
                    class_and_config_for_serialized_keras_object
                        raise ValueError(
                    ValueError: Unknown layer: BatchNormalization. Please ensure this object is passed to the `custom_objects` argument. See
                    https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
```

**To Reproduce**
Easiest way to reproduce is to edit the TF_MNIST example, steps are:
1. Change shape to '28, 28, 1' in mnist_shard_descriptor.py and delete the reshaping to 784 from download_data
2. Change shape in director config
3. Change network code to something like:
```def create_model():
    """Create keras model."""
      inputs = layers.Input((28, 28, 1))  
      conv_1 = layers.Conv2D(64, (3, 3),activation='relu')(inputs)  
      bn_1 = layers.BatchNormalization()(conv_1)  
      conv_2 = layers.Conv2D(64, (3, 3),activation='relu')(bn_1)  
      bn_2 = layers.BatchNormalization()(conv_2)  
      gap = layers.GlobalAveragePooling2D()(bn_2)  
      outputs = layers.Dense(10, name='predictions')(gap)  
      model = keras.Model(inputs=inputs, outputs=outputs)  
      return model
```
4. Observe the error on client while starting a collaborator service

**Expected behavior**
OpenFL should work with models implementing BN.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.0.2
 - Openfl version: 1.2.1
 - Tensorflow version: 2.8.0
 - Python version: 3.8.10
 - GPU: No

### PR DESCRIPTION
**Pull request by MasterSkepticista**
This closes #422.

Serialization of `tf.keras` models is natively supported as of https://github.com/keras-team/keras/pull/14748. 
This PR removes hotfix function to allow serialization on TF2.7 and greater.

### CODE REVIEW

**Comment by mansishr** [comment]
Hi @MasterSkepticista, thanks for linking a PR. I had talked to @psfoley regarding the issue and having a tensorflow version check before invoking the hotfix function might be a better way to go about it. This way, we can support tutorials having older tensorflow versions as well.

**Comment by MasterSkepticista** [comment]
@mansishr Version check is a possible solution.
More generally, I don't see any tutorials failing with the change so long as TF is above or equal 2.7

If there are existing outside projects using lower TF versions, a version check will be required.

**Comment by mansishr** [comment]
@MasterSkepticista that's right. I'll submit a PR with a version check before invoking the hotfix function. Thanks for your help!

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Tensorflow_MNIST.ipynb [hunk 1]
```diff
@@ -354,10 +496,203 @@
...
+    {
+     "data": {
+      "text/html": [
+       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">[13:00:16] </span><span style=\"color: #000080; text-decoration-color: #000080\">INFO    </span> Building <span style=\"color: #800000; text-decoration-color: #800000\">🡆</span> Object <span style=\"color: #800000; text-decoration-color: #800000\">CloudpickleSerializer</span> from <span style=\"color: #800000; text-decoration-color: #800000\">openfl.plugins.interface_serializer.cloudpickle_serializer</span> Module.                  <a href=\"file:///home/karan/workspace/part_bw/venv/lib/python3.8/site-packages/openfl/federated/plan/plan.py\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">plan.py</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///home/karan/workspace/part_bw/venv/lib/python3.8/site-packages/openfl/federated/plan/plan.py#171\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">171</span></a>\n",
```
Comment (psfoley) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Tensorflow_MNIST.ipynb]: @MasterSkepticista please clear cell output in the jupyter notebooks

**Comment by psfoley** [comment]
@MasterSkepticista A version check would be useful here for the reason that you mentioned; external projects may rely on an earlier TF version. @mansishr if you want to create a separate PR for that, then I think once cell outputs are cleared in the jupyter notebook this is ready to merge

**Comment by psfoley** [comment]
@MasterSkepticista could you please sign the CLA?

**Comment by MasterSkepticista** [comment]
***I have read the CLA Document and I hereby sign the CLA***

**Comment by mansishr** [comment]
@psfoley @MasterSkepticista I've created a PR #435 for TF version check.

**Pull request merged**